qiskit_experiments/framework/composite/composite_analysis.py

```diff
@@ -40,15 +37,79 @@ def _run_analysis(self, experiment_data: CompositeExperimentData, **options):
             QiskitError: if analysis is attempted on non-composite
                          experiment data.
         """
-        if not isinstance(experiment_data, CompositeExperimentData):
-            raise QiskitError("CompositeAnalysis must be run on CompositeExperimentData.")
+        composite_exp = experiment_data.experiment
+        component_exps = composite_exp.component_experiment()
+        if "component_job_metadata" in experiment_data.metadata:
```

Comment:
I don't understand these lines. What does the existence of `component_job_metadata` in the metadata mean, and why, if it exists, are you taking the last of `experiment_data.metadata["component_job_metadata"][-1]` ?  Please write inline documentation to explain.

Reply:
If I understand correctly the `component_job_metadata` is another set of metadata to initialize child experiment data. This is generated per job run so this means the last set of children. The metadata also contains `job_metadata` but this represents configurations of a full set of composite experiment (i.e. parent) so this doesn't help analysis. I don't think `job_metadata` is really necessary for composite experiment (more specifically, only run options is necessary to reconstruct experiment because other configs are usually unique to children).

Reply:
The important point we need to understand here is _children is not generated by job execution_, but this is executed by analysis. This is because experiment data child is not limited only to the composite, but also, for example, we can create pseudo RDB of experiment data. i.e. 
```python
my_all_result_for_writing_prx_paper = ExperimentData()
my_all_result_for_writing_prx_paper.add_child_data(my_data_of_calibration)
my_all_result_for_writing_prx_paper.add_child_data(my_data_of_rb)
my_all_result_for_writing_prx_paper.add_child_data(my_data_of_quantum_volume)
etc...
```
then we can write `my_all_result_for_writing_prx_paper` id in paper for reproducibility.

Reply:
This is to deal with how regular ExperimentData stores job metadata, which is used by some analysis (curve analysis in particular i think). If its not there certain experiments analysis wont run (note I think this should be changed, but it's beyond the scope of this PR).

Since each component experiment might need this for its respective analysis to work this is just storing the list of these component job metadata in the parent experiment so it can be added later along with the marginalized data. Previously this was all done when calling `run` which initialized all the child containers, but this is just moving it to analysis.

The `[-1]` was because this structure is a list for each run job, and you want the most recent one (same as job_metadata). Since #463 this isn't really necessary since you can't re-run more experiment jobs into the same container.